src/core/manager.c

```diff
@@ -2749,6 +2749,10 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                 r = unit_deserialize(u, f, fds);
                 if (r < 0)
                         goto finish;
+
+                r = unit_realize_cgroup(u);
```

Comment:
> Hence I think we should find a way to fix the original issue without changing this.

The obvious/naive solution would be to save/restore the value of `u->cgroup_realized_mask` as well as `u->cgroup_enabled_mask`.